sotodlib/site_pipeline/calibration/wiregrid.py

```diff
@@ -0,0 +1,434 @@
+        'limitswitchL1': _telescope + '.wg-actuator.feeds.wgactuator.limitswitch_LSL1',
+        'limitswitchL2': _telescope + '.wg-actuator.feeds.wgactuator.limitswitch_LSL2',
+        'limitswitchR1': _telescope + '.wg-actuator.feeds.wgactuator.limitswitch_LSR1',
+        'limitswitchR2': _telescope + '.wg-actuator.feeds.wgactuator.limitswitch_LSR2',
+        }
+    hk_dir = '/so/level2-daq/' + _telescope + '/hk'
+    #
+    _gl_aman = hk_utils.get_detcosamp_hkaman(tod,
+                                     alias = [_i for _i in alias_dict.keys()],
+                                     fields = [_j for _j in alias_dict.values()],
+                                     data_dir = hk_dir)
+    _gl_data = {}
+    for _i in alias_dict.keys():
+        if _i == 'enc_rad':
+            _gl_data[_i] = _gl_aman['wg-encoder']['wg-encoder'][0]
+            pass
+        else:
+            _gl_data[_i] = _gl_aman.restrict('hklabels_wg-actuator', [_i], in_place=False)['wg-actuator']['wg-actuator'][0]
+            pass
+    #
+    _ax_wg = core.AxisManager(tod.samps, tod.dets)
+    _ax_wg.wrap('enc_rad', _gl_data['enc_rad'] * wg_counts2rad, [(0,'samps')])
+    _ax_wg.wrap('LSL1', _gl_data['limitswitchL1'], [(0,'samps')])
+    _ax_wg.wrap('LSL2', _gl_data['limitswitchL2'], [(0,'samps')])
+    _ax_wg.wrap('LSR1', _gl_data['limitswitchR1'], [(0,'samps')])
+    _ax_wg.wrap('LSR2', _gl_data['limitswitchR2'], [(0,'samps')])
+    tod.wrap('wg', _ax_wg)
+    return tod
+
+def _get_operation_range(tod):
+    """
+    Define the range of the wire grid operation by the limit switches
+    """
+    idx_wg_inside = np.where(tod.wg.LSR2.astype(int) | tod.wg.LSL2.astype(int))[0]
+    return idx_wg_inside
+
+# Correct wires' direction for each telescope
+def correct_wg_angle(tod, telescope=None):
+    """
+    Correct offset of wires' direction by the mechanical design and hardware testing. This function is still under construction.
+
+    Parameters
+    ----------
+        tod : AxisManager
+        telescope : telescope type, e.g. satp1, satp3, etc.
+            this parameter will basically be filled by the obs_info wrapped in the axismanager.
+
+    Returns
+    -------
+        (tod, idx_wg_inside) : Tuple
+            tod is an AxisManager, and idx_wg_inside is the flags that indicates the opration range of the wire grid.
+    """
+    if telescope is None:
+        try:
+            _telescope = tod.obs_info.telescope
+        except:
+            print("Cannot load the telescope info. Please assign your target telescope")
+            logger.info("Telescope assignment was failed in site_pipeline.calibrate.wiregrid.correct_wg_angle")
+    tod = _wrap_wg_hk(tod)
+    idx_wg_inside = _get_operation_range(tod)
+    #
+    if telescope == 'satp1': wg_offset = np.deg2rad(12.13)
+    if telescope == 'satp2': wg_offset = np.deg2rad(9.473)
+    if telescope == 'satp3': wg_offset = np.deg2rad(11.21)
+    tod.wg.enc_rad = -  tod.wg.enc_rad + wg_offset
```

Comment:
It would probably be a good idea to store the _raw_ and _corrected_ values under different names.  E.g. `tod.wg.enc_rad = -tod.wg.enc_rad_raw + wg_offset`.  Will be less confusing, and that is "idempotent" (you can apply the correction multiple times and get the same result).